force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -63,20 +134,81 @@ public with sharing class DummyJSONCallout {
      * @param String jsonResponse The JSON string from the external system that contains user details.
      * @return Contact The Salesforce Contact object populated with details from the JSON string.
      */
-	@TestVisible // Allows test class to see this method. Since it is private, it would not be visible otherwise.
-	private static Contact createContactFromJson(String jsonResponse) {
-		// Deserialize the JSON string into map of primitive data types.
-
-		// Create a new contact from the JSON response.
-
-		// Set the contact fields from the JSON response.
-
-		// Deserialize the address from the JSON response.
-
-		// Set the address fields on the contact.
 
-		return null;
-	}
+	 @TestVisible
+	 private static Contact createContactFromJson(String jsonResponse) {
+		 // Log the raw JSON response
+		 System.debug('Raw JSON Response: ' + jsonResponse);
+		 
+		 // Deserialize the JSON string into a map of primitive data types
+		 Map<String, Object> responseMap = (Map<String, Object>) JSON.deserializeUntyped(jsonResponse);
+		 System.debug('Deserialized responseMap: ' + responseMap);
+		 
+		 // Create a new contact object
+		 Contact contact = new Contact();
+		 
+		 // Set the contact fields from the JSON response, ensure values exist
+		 if (responseMap.containsKey('email')) {
+			 contact.Email = (String) responseMap.get('email');
+		 } else {
+			 System.debug('Email not found in JSON response.');
```

Comment:
I would suggest doing this with the ternary operator like you did in the other method. The debug statements here are really not needed unless you are watching the logs for this. They are just making your code longer.